// random-profile-generator/dist/index.js
'use strict';

var _chance = require('chance');

var _chance2 = _interopRequireDefault(_chance);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _uniqueRandomArray = require('unique-random-array');

var _uniqueRandomArray2 = _interopRequireDefault(_uniqueRandomArray);

var _names = require('../data/names.json');

var _names2 = _interopRequireDefault(_names);

var _addresses = require('../data/addresses.json');

var _addresses2 = _interopRequireDefault(_addresses);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var chance = new _chance2.default();
var avatarBase = 'https://api.adorable.io/avatars/200/';

function profile() {
    var profile = {};
    var fullName = (0, _uniqueRandomArray2.default)(_names2.default)();
    var bDay = (0, _moment2.default)(chance.birthday());
    var today = (0, _moment2.default)(new Date());

    profile.fullName = fullName;
    profile.firstName = fullName.split(' ')[0];
    profile.lastName = fullName.split(' ')[1];
    profile.birthday = bDay.format('MMM Do[,] YYYY');
    profile.startDate = bDay.format('YYYY-MM-DD');
    profile.age = today.diff(bDay, 'years');
    profile.avatar = avatarBase + fullName.replace(/ /g, '-');
    profile.address = (0, _uniqueRandomArray2.default)(_addresses2.default)();
    profile.phone = chance.phone();
    profile.email = chance.email();
    profile.twitter = chance.twitter();
    profile.ssn = chance.ssn();

    return profile;
}

function age() {
    return chance.age();
}

function birthday() {
    return (0, _moment2.default)(chance.birthday()).format('MMM Do[,] YYYY');
}

function avatar() {
    return avatarBase + (0, _uniqueRandomArray2.default)(_names2.default)().replace(/ /g, '-');
}

function phone() {
    return chance.phone();
}

function email() {
    return chance.email();
}

function twitter() {
    return chance.twitter();
}

function ssn() {
    return chance.ssn();
}

module.exports = {
    profile: profile,
    name: (0, _uniqueRandomArray2.default)(_names2.default),
    address: (0, _uniqueRandomArray2.default)(_addresses2.default),
    birthday: birthday,
    age: age,
    avatar: avatar,
    phone: phone,
    email: email,
    twitter: twitter,
    ssn: ssn
};
